extensions/interactions/RatioExpressionInput/directives/ratio-expression-input-validation.service.spec.ts

```diff
@@ -211,6 +220,34 @@ describe('RatioExpressionInputValidationService', () => {
       message: 'Rule 3 from answer group 1 will never be matched because' +
       ' it has differing number of terms than required.'
     }]);
+
+    // The second rule will never get matched.
+    answerGroups[0].rules = [hasSpecificTermEqualTo, equals];
+
+    warnings = validatorService.getAllWarnings(
+      currentState, customizationArgs, answerGroups, goodDefaultOutcome);
+    expect(warnings).toEqual([{
+      type: WARNING_TYPES.ERROR,
+      message: 'Rule 2 from answer group 1 will never be matched because ' +
+      'it is preceded by a \'HasSpecificTermEqualTo\' rule with a ' +
+      'matching input.'
+    }]);
+
+    let invalidHasSpecificTermEqualTo = rof.createFromBackendDict({
+      rule_type: 'HasSpecificTermEqualTo',
+      inputs: {
+        x: 5, y: 1
+      }
+    }, 'RatioExpressionInput');
```

Comment:
I think we should also add corner cases where the term position is equal to the expected number of terms, and also where it's just one more than the expected number of terms.

Let's also add a case where the expected number of terms is set to 0 (which implies that a ratio of any length will be accepted.

Reply:
Good point.

- Changed `invalidHasSpecificTermEqualTo` to have the 4th term equal to 1 (instead of 5th term equal to 1) in order to test where it's just one more than the expected number of terms (3). 
- Added a `validHasSpecificTermEqualTo` case
- Added a expected number of terms = 0 case